recipes/powershell/storage/replication_target_on_dv_operations_end_to_end.ps1

```diff
@@ -0,0 +1,429 @@
+                        -Body ($add_rep_tgt_json)  `
+                        -ContentType $contentType  `
+                        -Headers $headers
+    
+    if ($response_add_rep_target.StatusCode -ne 204)
+    {
+        throw "Unable to add replication target." 
+    }
+    
+    Write-Host "replication target added successfully.`n"
+    echo $response_add_rep_target
+    Write-Host $response_add_rep_target
+    
+    $response_add_rep_target = (ConvertFrom-Json -InputObject $response_add_rep_target)
+    retrun $response_add_rep_target
+}
+
+#################################################
+# Get all replication target details
+#################################################
+Function GetAllReplicationTargets([string]$stsid, [string]$dvid)
+{
+    $get_rep_tgt_uri = $baseUri + "/storage/storage-servers" + $stsid + "/disk-volumes/" + $dvid | Format-Hex + "/replication-targets"
+    
+    $response_get_rep_target = Invoke-WebRequest   `
+                        -Uri $get_rep_tgt_uri      `
+                        -Method GET                `
+                        -ContentType $contentType  `
+                        -Headers $headers
+    
+    if ($response_get_rep_target.StatusCode -ne 200)
+    {
+        throw "Unable to get replication targets." 
+    }
+    
+    Write-Host "replication target information recieved successfully.`n"
+    echo $response_get_rep_target
+    Write-Host $response_get_rep_target
+    
+    $response_get_rep_target = (ConvertFrom-Json -InputObject $response_get_rep_target)
+}
+
+#################################################
+# Get all replication target details for a specific replication target
+#################################################
+Function GetReplicationTargetById([string]$stsid, [string]$dvid, [string]$reptargetid)
+{
+    $get_rep_tgt_uri = $baseUri + "/storage/storage-servers" + $stsid + "/disk-volumes/" + $dvid | Format-Hex + "/replication-targets/" + reptargetid
+    
+    $response_get_rep_target = Invoke-WebRequest   `
+                        -Uri $get_rep_tgt_uri      `
+                        -Method GET                `
+                        -ContentType $contentType  `
+                        -Headers $headers
+    
+    if ($response_get_rep_target.StatusCode -ne 200)
+    {
+        throw "Unable to get replication targets." 
+    }
+    
+    Write-Host "replication target information recieved successfully.`n"
+    echo $response_get_rep_target
+    Write-Host $response_get_rep_target
+    
+    $response_get_rep_target = (ConvertFrom-Json -InputObject $response_get_rep_target)
+}
+
+#################################################
+# Delete replication target details
+#################################################
+Function DeleteReplicationTargets([string]$stsid, [string]$dvid)
+{
+    $del_rep_tgt_uri = $baseUri + "/storage/storage-servers" + $stsid + "/disk-volumes/" + $dvid | Format-Hex + "/replication-targets"
+    
+    $del_rep_tgt_json = '{
+      {
```

Comment:
This isn't valid JSON. The first brace introduces an object, so the second brace needs an attribute name before it.